pkg/userserver/user_server.go

```diff
@@ -163,8 +228,12 @@ func (k *userServer) ServeHTTP(wr http.ResponseWriter, req *http.Request) {
 	}
 
 	// get service proxy host for current managed cluster
+	namespace, ok := k.addonNamespaceMapper[tsc.Cluster]
+	if !ok {
```

Comment:
Using only informer, lister read(not using mapper) would be simpler.

No election would be better.

cc @qiujian16